docs/src/index.md

```diff
@@ -1,12 +1,377 @@
-# GLM.jl Documentation
+# GLM Documentation
 
 ```@meta
 DocTestSetup = quote
-    using Distributions, GLM
+    using CategoricalArrays, DataFrames, Distributions, GLM, RDatasets
 end
 ```
 
-## Types defined in the package
+## Installation
+
+```julia
+Pkg.add("GLM")
+```
+
+will install this package and its dependencies, which includes the [Distributions package](https://github.com/JuliaStats/Distributions.jl).
+
+The [RDatasets package](https://github.com/johnmyleswhite/RDatasets.jl) is useful for fitting models on standard R datasets to compare the results with those from R.
+
+## Fitting GLM models
+
+To fit a Generalized Linear Model (GLM), use the function, `glm(formula, data, family, link)`, where,
+- `formula`: uses column symbols from the DataFrame data, for example, if `names(data)=[:Y,:X1,:X2]`, then a valid formula is `@formula(Y ~ X1 + X2)`
+- `data`: a DataFrame which may contain NA values, any rows with NA values are ignored
+- `family`: chosen from `Bernoulli()`, `Binomial()`, `Gamma()`, `Normal()`, or `Poisson()`
+- `link`: chosen from the list below, for example, `LogitLink()` is a valid link for the `Binomial()` family
+
+An intercept is included in any GLM by default.
+
+## Methods applied to fitted models
+
+Many of the methods provided by this package have names similar to those in [R](http://www.r-project.org).
+- `coef`: extract the estimates of the coefficients in the model
+- `deviance`: measure of the model fit, weighted residual sum of squares for lm's
+- `dof_residual`: degrees of freedom for residuals, when meaningful
+- `glm`: fit a generalized linear model (an alias for `fit(GeneralizedLinearModel, ...)`)
+- `lm`: fit a linear model (an alias for `fit(LinearModel, ...)`)
+- `stderror`: standard errors of the coefficients
+- `vcov`: estimated variance-covariance matrix of the coefficient estimates
+- `predict` : obtain predicted values of the dependent variable from the fitted model
+
+## Minimal examples
+
+### Ordinary Least Squares Regression:
+```jldoctest
+julia> using DataFrames, GLM
+
+julia> data = DataFrame(X=[1,2,3], Y=[2,4,7])
+3×2 DataFrames.DataFrame
+│ Row │ X │ Y │
+├─────┼───┼───┤
+│ 1   │ 1 │ 2 │
+│ 2   │ 2 │ 4 │
+│ 3   │ 3 │ 7 │
+
+julia> ols = lm(@formula(Y ~ X), data)
+StatsModels.DataFrameRegressionModel{GLM.LinearModel{GLM.LmResp{Array{Float64,1}},GLM.DensePredChol{Float64,Base.LinAlg.Cholesky{Float64,Array{Float64,2}}}},Array{Float64,2}}
+
+Formula: Y ~ 1 + X
+
+Coefficients:
+              Estimate Std.Error  t value Pr(>|t|)
+(Intercept)  -0.666667   0.62361 -1.06904   0.4788
+X                  2.5  0.288675  8.66025   0.0732
+
+julia> stderror(ols)
+2-element Array{Float64,1}:
+ 0.62361
+ 0.288675
+
+julia> predict(ols)
+3-element Array{Float64,1}:
+ 1.83333
+ 4.33333
+ 6.83333
+
+```
+<!-- Andreas Noack: As of 9 May 2018 this example doesn't work so I've temporarily commented it out
```

Comment:
Nice, I always forget that markdown supports HTML-style comments.

Reply:
the original and still best